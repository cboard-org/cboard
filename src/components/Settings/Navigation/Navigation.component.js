import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Switch from '@material-ui/core/Switch';
import Paper from '@material-ui/core/Paper';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import FullScreenDialog from '../../UI/FullScreenDialog';
import messages from './Navigation.messages';

import './Navigation.css';

const propTypes = {
  /**
   * Callback fired when clicking the back button
   */
  onClose: PropTypes.func,
  updateNavigationSettings: PropTypes.func.isRequired,
  navigationSettings: PropTypes.object.isRequired
};

class Navigation extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      ...props.navigationSettings
    };
  }

  toggleNavigation = () => {
    this.setState({
      active: !this.state.active
    });
  };

  onSubmit = () => {
    this.props.updateNavigationSettings(this.state);
  };

  render() {
    const { onClose } = this.props;
    return (
      <div className="Navigation">
        <FullScreenDialog
          open
          title={<FormattedMessage {...messages.navigation} />}
          onClose={onClose}
          onSubmit={this.onSubmit}
        >
          <Paper>
            <List>
              <ListItem>
                <ListItemText
                  primary={<FormattedMessage {...messages.enable} />}
                  secondary={<FormattedMessage {...messages.enableSecondary} />}
                />
                <ListItemSecondaryAction>
                  <Switch
                    checked={this.state.active}
                    onChange={this.toggleNavigation}
                    value="active"
                    color="primary"
                  />
                </ListItemSecondaryAction>
              </ListItem>
            </List>
          </Paper>
        </FullScreenDialog>
      </div>
    );
  }
}

Navigation.propTypes = propTypes;

export default Navigation;
