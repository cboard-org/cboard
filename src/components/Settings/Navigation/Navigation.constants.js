export const NAVIGATION_BUTTONS_STYLE_SIDES = 'sides';
export const NAVIGATION_BUTTONS_STYLE_TOP = 'top';

export const NAVIGATION_BUTTONS_STYLES = [
  { name: 'onTheSides', value: 'sides' },
  { name: 'onTop', value: 'top' }
];
