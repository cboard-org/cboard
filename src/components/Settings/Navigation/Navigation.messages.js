import { defineMessages } from 'react-intl';

export default defineMessages({
  navigation: {
    id: 'cboard.components.Settings.Navigation.navigation',
    defaultMessage: 'Navigation'
  },
  enable: {
    id: 'cboard.components.Settings.Navigation.enable',
    defaultMessage: 'Enable context aware back button'
  },
  enableSecondary: {
    id: 'cboard.components.Settings.Navigation.enableSecondary',
    defaultMessage: 'Shows big back buttons on top of the boards'
  }
});
