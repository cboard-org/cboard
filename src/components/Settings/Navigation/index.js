export { default } from './Navigation.container';
