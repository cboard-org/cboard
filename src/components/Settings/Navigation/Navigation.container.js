import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { injectIntl, intlShape } from 'react-intl';
import { updateNavigationSettings } from '../../App/App.actions';
import Navigation from './Navigation.component';

export class NavigationContainer extends PureComponent {
  static propTypes = {
    intl: intlShape.isRequired
  };

  render() {
    const { history } = this.props;

    return <Navigation onClose={history.goBack} {...this.props} />;
  }
}

NavigationContainer.props = {
  history: PropTypes.object,
  updateNavigationSettings: PropTypes.func.isRequired,
  navigationSettings: PropTypes.object.isRequired
};

const mapStateToProps = ({ app: { navigationSettings } }) => ({
  navigationSettings
});
const mapDispatchToProps = { updateNavigationSettings };

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(NavigationContainer));
