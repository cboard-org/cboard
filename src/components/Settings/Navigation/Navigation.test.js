import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Navigation from './Navigation.component';

jest.mock('./Navigation.messages', () => {
  return {
    navigation: {
      id: 'cboard.components.Settings.Navigation.navigation',
      defaultMessage: 'Navigation'
    },
    enable: {
      id: 'cboard.components.Settings.Navigation.enable',
      defaultMessage: 'Enable text...'
    },
    enableSecondary: {
      id: 'cboard.components.Settings.Navigation.enableSecondary',
      defaultMessage: 'some text secondary'
    }
  };
});

const INITIAL_NAVIGATION_SETTINGS = {
  active: false
};

let navigationSettings = INITIAL_NAVIGATION_SETTINGS;

const COMPONENT_PROPS = {
  navigationSettings,
  updateNavigationSettings: payload => {
    navigationSettings = { ...navigationSettings, ...payload };
  },
  onClose: () => {}
};

describe('Navigation tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<Navigation {...COMPONENT_PROPS} />);
  });

  test('switch behavior', () => {
    const wrapper = shallow(<Navigation {...COMPONENT_PROPS} />);
    let tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();

    const state = wrapper.state();

    const switchElement = wrapper.find('WithStyles(Switch)').at(0);
    switchElement.simulate('change');

    const newState = wrapper.state();

    expect(state.active).not.toBe(newState.active);

    tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
