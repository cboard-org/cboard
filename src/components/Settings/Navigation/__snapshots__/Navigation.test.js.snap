// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation tests default renderer 1`] = `
<div
  className="Navigation"
>
  <WithStyles(FullScreenDialog)
    onClose={[Function]}
    onSubmit={[Function]}
    open={true}
    title={
      <FormattedMessage
        defaultMessage="Navigation"
        id="cboard.components.Settings.Navigation.navigation"
        values={Object {}}
      />
    }
  >
    <WithStyles(Paper)>
      <WithStyles(List)>
        <WithStyles(ListItem)>
          <WithStyles(ListItemText)
            primary={
              <FormattedMessage
                defaultMessage="Enable text..."
                id="cboard.components.Settings.Navigation.enable"
                values={Object {}}
              />
            }
            secondary={
              <FormattedMessage
                defaultMessage="some text secondary"
                id="cboard.components.Settings.Navigation.enableSecondary"
                values={Object {}}
              />
            }
          />
          <WithStyles(ListItemSecondaryAction)>
            <WithStyles(Switch)
              checked={false}
              color="primary"
              onChange={[Function]}
              value="active"
            />
          </WithStyles(ListItemSecondaryAction)>
        </WithStyles(ListItem)>
      </WithStyles(List)>
    </WithStyles(Paper)>
  </WithStyles(FullScreenDialog)>
</div>
`;

exports[`Navigation tests switch behavior 1`] = `
<div
  className="Navigation"
>
  <WithStyles(FullScreenDialog)
    onClose={[Function]}
    onSubmit={[Function]}
    open={true}
    title={
      <FormattedMessage
        defaultMessage="Navigation"
        id="cboard.components.Settings.Navigation.navigation"
        values={Object {}}
      />
    }
  >
    <WithStyles(Paper)>
      <WithStyles(List)>
        <WithStyles(ListItem)>
          <WithStyles(ListItemText)
            primary={
              <FormattedMessage
                defaultMessage="Enable text..."
                id="cboard.components.Settings.Navigation.enable"
                values={Object {}}
              />
            }
            secondary={
              <FormattedMessage
                defaultMessage="some text secondary"
                id="cboard.components.Settings.Navigation.enableSecondary"
                values={Object {}}
              />
            }
          />
          <WithStyles(ListItemSecondaryAction)>
            <WithStyles(Switch)
              checked={false}
              color="primary"
              onChange={[Function]}
              value="active"
            />
          </WithStyles(ListItemSecondaryAction)>
        </WithStyles(ListItem)>
      </WithStyles(List)>
    </WithStyles(Paper)>
  </WithStyles(FullScreenDialog)>
</div>
`;

exports[`Navigation tests switch behavior 2`] = `
<div
  className="Navigation"
>
  <WithStyles(FullScreenDialog)
    onClose={[Function]}
    onSubmit={[Function]}
    open={true}
    title={
      <FormattedMessage
        defaultMessage="Navigation"
        id="cboard.components.Settings.Navigation.navigation"
        values={Object {}}
      />
    }
  >
    <WithStyles(Paper)>
      <WithStyles(List)>
        <WithStyles(ListItem)>
          <WithStyles(ListItemText)
            primary={
              <FormattedMessage
                defaultMessage="Enable text..."
                id="cboard.components.Settings.Navigation.enable"
                values={Object {}}
              />
            }
            secondary={
              <FormattedMessage
                defaultMessage="some text secondary"
                id="cboard.components.Settings.Navigation.enableSecondary"
                values={Object {}}
              />
            }
          />
          <WithStyles(ListItemSecondaryAction)>
            <WithStyles(Switch)
              checked={true}
              color="primary"
              onChange={[Function]}
              value="active"
            />
          </WithStyles(ListItemSecondaryAction)>
        </WithStyles(ListItem)>
      </WithStyles(List)>
    </WithStyles(Paper)>
  </WithStyles(FullScreenDialog)>
</div>
`;
